### PR TITLE
Snyk Mock Server Test

### PR DESCRIPTION
**Pull request by japurva1502**
Signed-off-by: Apurva <apurva.jain@gatech.edu>

### CODE REVIEW

File: src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        Component component = new Component();
+        component.setPurl(new PackageURL("pkg:maven/com.fasterxml.woodstox/woodstox-core@5.0.0"));
+        component.setPurlCoordinates("pkg:maven/com.fasterxml.woodstox/woodstox-core@5.0.0");
+        List<Component> list = new ArrayList<>();
+        list.add(component);
+        SnykAnalysisTask task = new SnykAnalysisTask();
+        SnykAnalysisEvent event = new SnykAnalysisEvent(component);
+        task.inform(event);
+        //task.apiBaseUrl = "http://localhost:1080";
+        //task.analyze(list);
```
Comment (nscuro) [inline src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java:100]: Please remove.

File: src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            Assert.assertEquals(Severity.CRITICAL, vulnerability.getSeverity());
+            Assert.assertNotNull(vulnerability.getCreated());
+            Assert.assertNotNull(vulnerability.getAliases());
+            Assert.assertEquals(1, vulnerability.getAliases().size());
+            Assert.assertEquals("A", vulnerability.getAliases().get(0).getCveId());
+        };
+
+        Vulnerability vulnerability = qm.getVulnerabilityByVulnId("SNYK", "SNYK-JAVA-COMFASTERXMLWOODSTOX-2928754", true);
+        assertVulnerability.accept(vulnerability);
+
```
Comment (nscuro) [inline src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java:117]: There should be an assertion that the component `component` has been marked as vulnerable to `vulnerability`. You can use [`qm.contains(Vulnerability, Component)`](https://github.com/DependencyTrack/dependency-track/blob/master/src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java#L261-L278) for this.

**Review by nscuro — CHANGES_REQUESTED** [review]

File: src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java (hunk 2 above)
Comment (japurva1502) [reply src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java:122]: Done

File: src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java (hunk 1 above)
Comment (japurva1502) [reply src/test/java/org/dependencytrack/tasks/scanners/SnykMockServerTest.java]: Done

**Pull request closed without merging**